### PR TITLE
refactor status-uploader

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
refactor status-uploader, so that github assets of multiple testruns can be uploaded at once. Also archive assets and provide an overview file. 

**Release note**:

```
refactor status-uploader, so that github assets of multiple testruns can be uploaded at once. Also archive assets and provide an overview file. 

```

### CODE REVIEW

File: pkg/testrunner/result/status-uploader.go [hunk 1]
```diff
@@ -179,23 +406,86 @@ func getGithubClient(repoURL *url.URL, githubUser, githubPassword string) (*gith
...
+type ComponentExtended struct {
+	PlainURL        string
+	Version         string
+	GithubClient    *github.Client
+	Owner           string
+	Name            string
+	GithubReleaseID int64
+}
+
+func parseComponent(component *componentdescriptor.Component, githubUser string, githubPassword string) (ComponentExtended, error) {
```
Comment (schrodit) [inline pkg/testrunner/result/status-uploader.go]: some documentation what the function is doing and what a extended component is

File: pkg/testrunner/result/status-uploader.go [hunk 2]
```diff
@@ -11,137 +12,354 @@ import (
...
+	if err != nil {
+		l.Error(err, "failed to parse components")
+		return err
 	}
-	tableString := strings.Builder{}
-	output.RenderStatusTable(&tableString, tr.Status.Steps)
-	statusOutput := fmt.Sprintf("Testrun: %s\n\n%s\n%s", tr.Name, tableString.String(), util.PrettyPrintStruct(tr.Status))
-	filenameWithoutStatus := fmt.Sprintf("%s%s-%s-%s", assetPrefix, md.Landscape, md.CloudProvider, md.KubernetesVersion)
-	filename := fmt.Sprintf("%s-%s.txt", filenameWithoutStatus, testrunStatusTag)
+	randomComponent := extendedComponents[0]
```
Comment (schrodit) [inline pkg/testrunner/result/status-uploader.go]: why use a random component?

File: pkg/testrunner/result/status-uploader.go [hunk 3]
```diff
@@ -11,137 +12,354 @@ import (
...
-	repoURL, err := url.Parse(fmt.Sprintf("https://%s", component.Name))
-	if err != nil {
+	testrunsToUpload, err := identifyTestrunsToUpload(runs, randomComponent, overviewFilepath)
+	if testrunsToUpload == nil || len(*testrunsToUpload) == 0 {
+		l.Info("no testrun updates, therefore not assets to upload")
+		return nil
+	}
+	l.Info(fmt.Sprintf("identified %d testruns for github asset upload", len(*testrunsToUpload)))
+
+	archiveFilename := fmt.Sprintf("%s%s.zip", prefix, (*runs)[0].Metadata.Landscape)
```
Comment (schrodit) [inline pkg/testrunner/result/status-uploader.go]: do we want `zip` or `tar.gz` what do you think is more convenient?

File: pkg/testrunner/result/status-uploader.go (hunk 1 above)
Comment (OlegLoewen) [reply pkg/testrunner/result/status-uploader.go]: Will add documentation: wraps component struct with additional github properties: github client, repo owner, repo name, release ID.

**Pull request merged**